### ISSUE TITLE
RFE: normalize AUDIT_NETFILTER_PKT events

### ISSUE THREAD

**Issue by pcmoore** [linux-audit/audit-kernel#11]
The AUDIT_NETFILTER_PKT audit events are not normalized. They swing fields in and out based on settings rather than changing the value of the event. Here's one example in ./net/netfilter/xt_AUDIT.c:

```
if (ntohs(ih->frag_off) & IP_OFFSET) {
    audit_log_format(ab, " frag=1");
    return;
}
```

frag should always be set like:

```
audit_log_format(ab, " frag=%d", ntohs(ih->frag_off) & IP_OFFSET);
```

**Comment by pcmoore** [comment]
Older upstream discussion regarding AUDIT_NETFILTER_PKT events:

* https://www.redhat.com/archives/linux-audit/2013-February/msg00020.html

**Comment by rgbriggs** [comment]
Posted a query to linux-audit and netfilter-devel mailing lists requesting some guidance on how to proceed with pulling in fields and how many combinations are acceptable for one message type:
  https://www.redhat.com/archives/linux-audit/2017-January/msg00074.html

**Comment by rgbriggs** [comment]
Guidance received, working on design of number of different messages or combined message.

**Comment by rgbriggs** [comment]
Posted proposed message format with some length stats and more questions:
https://www.redhat.com/archives/linux-audit/2017-January/msg00092.html

**Comment by rgbriggs** [comment]
Posted preliminary test framework patch for comment:
https://github.com/rgbriggs/audit-testsuite/tree/ghak11-AUDIT_NETFILTER_PKT-normalization

**Comment by pcmoore** [comment]
> Posted preliminary test framework patch for comment:
> https://github.com/rgbriggs/audit-testsuite/tree/ghak11-AUDIT_NETFILTER_PKT-normalization

I realize this is still just an early draft, but did you mean to submit this as a PR against the audit-testsuite?

**Comment by rgbriggs** [comment]
Not yet.  Just looking for some initial feedback that I'm on the right path...

**Comment by rgbriggs** [comment]
Posted RFC patch for message format upstream on linux-audit and netfilter-devel lists:
https://www.redhat.com/archives/linux-audit/2017-January/msg00118.html

**Comment by pcmoore** [comment]
@rgbriggs okay, I see the mailing list posting, I guess I'm just trying to figure out that previous comment here ... can we comment on branches if they are not PRs? (It wasn't immediately obvious to me)  Or was this just a FYI that a mailing list post was due very soon?

Any reason is fine - I've never complained about someone being to verbose when it comes to status reporting - I just want to make sure I'm clear on what you are expecting from me/us :)

**Comment by rgbriggs** [comment]
It wasn't intended as an FYI that a mailing list posting was due.  It was more wondering aloud if anyone knew how to comment on a push to a personal fork without making a PR.  If you know how, I'd like to know how to do so.

Aside from that, I'm looking for feedback on that testsuite patch either on that personal fork in github, or on the mailing list testsuite patch RFC.

**Comment by rgbriggs** [comment]
Message format simplified considerably to include only:
  mark, saddr, daddr, proto

Patch v2 posted upstream Cc:netfilter-devel
   https://www.redhat.com/archives/linux-audit/2017-February/msg00146.html

**Comment by rgbriggs** [comment]
Original requirements for RHEL6.1, 2010: https://bugzilla.redhat.com/show_bug.cgi?id=642391

**Comment by rgbriggs** [comment]
Patch v3 posted upstream Cc: netfilter-devel
https://www.redhat.com/archives/linux-audit/2017-February/msg00170.html

**Comment by pcmoore** [comment]
Patch v4 posted upstream: https://www.redhat.com/archives/linux-audit/2017-March/msg00123.html

**Comment by pcmoore** [comment]
Merged into audit/next via 36fe46d172e52a37bf9293f130610129c1239f56.
